File: test/fixtures/ruby/post_binary_file.rb
require 'net/http'

uri = URI('http://localhost:28139/american-art/query')
req = Net::HTTP::Post.new(uri)
req.content_type = 'application/sparql-query'
req['Accept'] = 'application/sparql-results+json'

req.body = File.binread('./sample.sparql').delete("\n")

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/php/post_binary_file.php
<?php
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'http://localhost:28139/american-art/query');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Content-type: application/sparql-query',
    'Accept: application/sparql-results+json',
]);
curl_setopt($ch, CURLOPT_POSTFIELDS, file_get_contents('./sample.sparql'));

$response = curl_exec($ch);

curl_close($ch);
